tmt/steps/__init__.py

```diff
@@ -206,6 +206,23 @@ def is_in_standalone_mode(self) -> bool:
         """
         return False
 
+    def assert_feeling_safe(self, deprecated_in_version: str, subject: str) -> None:
+        """
+        Raises a tmt.utils.ProvisionError if feeling-safe is required, but not set.
+        Warns when feeling-safe will be required in a future version.
+        :param deprecated_in_version: Version from which feeling-safe is required, e.g. '1.38'.
+        :param subject: Subject requiring feeling-safe, e.g. 'Local provision plugin'.
+        """
+        if self.is_feeling_safe:
+            return
+
+        if tmt.__version__ < deprecated_in_version:
+            self.warn(f"{subject} will require '--feeling-safe' option "
+                      f"from version {deprecated_in_version}.")
+
+        else:
+            raise tmt.utils.ProvisionError(f"{subject} requires '--feeling-safe' option")
```

Comment:
`GeneralError` would be probably better, more accepting than `ProvisionError` when `assert_feeling_safe()` gets called from non-`provision` plugins one day.